src/Stores/TaskStore.ts

```diff
@@ -0,0 +1,48 @@
+import { action, computed, makeObservable, observable } from "mobx";
+import TaskItem from "../Models/TaskItem";
+import { v4 as uuidv4 } from 'uuid';
+
+class TaskStore {
+
+    @observable tasks: TaskItem[] = observable.array([]);
+   
+    @action addTask = (taskItem: TaskItem) => {
+        debugger;        
+        this.tasks.push({...taskItem,id : this.tasks.length > 0? Math.max(...this.tasks.map(t=>t.id!)) + 1 : 1})        
```

Comment:
just to know you understand the concept, can you explain why did you use `!`  in `id!`

Reply:
try to simplify this code with your es6 learnings  :)

Reply:
why did you use `math.max` here, as i understand this is simple logic to add next `id`

Reply:
The id property is an optional property in my TaskItem class, therefore I assure to TS that in this line of code the id property will NOT be empty. 
"!" is definite assignment.

Reply:
> why did you use `math.max` here, as i understand this is simple logic to add next `id`

Instead I can use id : Math.random() or Date.now()

I have followed some method.

//Method 1      
        this.tasks.push({ ...taskItem, id: Math.random()})
        //Method 2
        //this.tasks.push({ ...taskItem, id: Date.now()})   

        //Method 3 use reduce  (Performance issue)          
        //this.tasks.push({ ...taskItem, id: this.tasks.reduce((max, item) => (item.id > max ? item.id : max),this.tasks[0].id) + 1 })

        //Method 4 spread operator and map
        //this.tasks.push({ ...taskItem, id: this.tasks.length > 0 ? Math.max(...this.tasks.map(t => t.id!)) + 1 : 1 })